detail/workload/index.vue

```diff
@@ -171,14 +180,53 @@ export default {
       const total = this.isCronJob ? this.totalRuns : this.value.pods.length;
 
       return !jobGauges.find(jg => jg.count === total);
-    }
+    },
+
+    canScale() {
+      return !!SCALABLE_TYPES.includes(this.value.type) && this.value.canUpdate;
+    },
   },
+  methods: {
+    async scale(isUp) {
```

Comment:
Would not be better to use simply async/await?
Also it does not make sense to use async/await twice.

```js
/**
 * Handle workload scale
 */
scale(isUp) {
  this.disabled = true;
  const request = isUp ? this.value.scaleUp : this.value.scaleDown

  return request()
    .catch(error => this.scaleError(isUp ? 'up' : 'down'))
    .finally(() => (this.disabled = false))
  )
}

/**
 * Display error pop up
 */
scaleError(direction) {
  this.$store.dispatch('growl/fromError', {
    title: this.t('workload.list.errorCannotScale', {
      direction,
      workloadName: this.value.name
    }),
    error
  })
}
```

Reply:
Refactored it to be like this:

```js
async scale(isUp) {
      try {
        if (isUp) {
          await this.value.scaleUp();
        } else {
          await this.value.scaleDown();
        }
      } catch (err) {
        this.$store.dispatch('growl/fromError', {
          title: this.t('workload.list.errorCannotScale', { direction: isUp ? 'up' : 'down', workloadName: this.value.name }),
          err
        },
        { root: true });
      }
    },
```